pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -331,13 +333,15 @@ func (configgen *ConfigGeneratorImpl) createGatewayHTTPFilterChainOpts(
 		// Build a filter chain for each HTTPS server
 		// We know that this is a HTTPS server because this function is called only for ports of type HTTP/HTTPS
 		// where HTTPS server's TLS mode is not passthrough and not nil
+		enVal := os.Getenv("ENABLE_INGRESS_GATEWAY_AGENT")
```

Comment:
See above - please use node metadata.

Reply:
Note that using node metadata also allows the reuse of the code for sidecars.

Reply:
This has been removed. Pilot will check node metadata instead of environment variable. Thanks for the advice.